### PR TITLE
Port groundfilter

### PR DESCRIPTION
**Pull request by keskitalo**
Support for ground (azimuth-locked) filtering, both full and split by scan direction.

Also small fixes to `ground_sim.py`.

### CODE REVIEW

File: src/toast/tests/ops_groundfilter.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+import os
+
+import numpy as np
+
+from ..tod import AnalyticNoise, OpSimNoise
+from ..todmap import OpGroundFilter, TODGround
+
+from ._helpers import create_outdir, create_distdata, boresight_focalplane
+"""
```
Comment (tskisner) [inline src/toast/tests/ops_groundfilter.py:37]: Probably safe to delete this now, correct?

File: src/toast/tests/ops_groundfilter.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+class GroundFilterTest(MPITestCase):
+    def setUp(self):
+        fixture_name = os.path.splitext(os.path.basename(__file__))[0]
+        self.outdir = create_outdir(self.comm, fixture_name)
+        np.random.seed(123456)
+        self.shared_flag_mask = 1
+
+    def test_groundfilter(self):
+
+        # Create a fake satellite data set for testing
```
Comment (tskisner) [inline src/toast/tests/ops_groundfilter.py:49]: "Create fake ground data ..."

File: src/toast/ops/sim_ground.py [hunk 3]
```diff
@@ -650,6 +652,20 @@ def _exec(self, data, detectors=None, **kwargs):
                 | ob.intervals[self.turn_rightleft_interval]
             )
 
+            # Get the Sun's position in horizontal coordinates and define
+            # "Sun up" and "Sun close" intervals according to it
+
+            add_solar_intervals(
```
Comment (tskisner) [inline src/toast/ops/sim_ground.py:658]: Thanks for adding this functionality!

File: src/toast/ops/groundfilter.py [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+        nfilter = self.filter_order + 1
+        legendre_templates = np.zeros([nfilter, phase.size])
+        legendre(phase, legendre_templates, 0, nfilter)
+        if not self.split_template:
+            legendre_filter = legendre_templates
+        else:
+            # Create separate templates for alternating scans
+            common_flags = obs.shared[self.shared_flags].data
+            legendre_filter = []
+            # The flag masks are hard-coded in sim_ground.py
```
Comment (tskisner) [inline src/toast/ops/groundfilter.py:173]: In the future, there will be many ways of populating Observations with both real and simulated data.  This is fine for now, but we will need to make the left and right-going flags / intervals be a configurable parameter eventually.  This goes beyond a simple port of toast2 code though, so fine to leave for now.

**Review by tskisner — APPROVED** [review]
Looks great, thanks.

File: src/toast/tests/ops_groundfilter.py (hunk 1 above)
Comment (keskitalo) [reply src/toast/tests/ops_groundfilter.py:37]: Thanks!

File: src/toast/tests/ops_groundfilter.py (hunk 2 above)
Comment (keskitalo) [reply src/toast/tests/ops_groundfilter.py:49]: Thanks!

File: src/toast/ops/sim_ground.py (hunk 3 above)
Comment (keskitalo) [reply src/toast/ops/sim_ground.py:658]: No problem!

File: src/toast/ops/groundfilter.py (hunk 4 above)
Comment (keskitalo) [reply src/toast/ops/groundfilter.py:173]: I agree.

**Pull request merged**